src/lib/ibcclient.ts

```diff
@@ -82,6 +83,20 @@ import {
   toIntHeight,
 } from './utils';
 
+function formatMessage<T>(
+  message: T,
+  mutateFn: (
+    deepClonedMessage: T,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    encode: (data: Uint8Array) => any
+  ) => void
+) {
+  const deepClonedMessage = cloneDeep(message);
+  mutateFn(deepClonedMessage, toBase64);
+
+  return deepClonedMessage;
```

Comment:
The type is inferred so it's `T` (the same as the input message), however, this is not exactly true because `encode` function changes `UInt8Array`s to `string`s.

Reply:
Then let's be explicit and return `unknown` or `any` or `Record<string, unknown>`.
Ideally the most restrictive type we can pass to this.logger.debug.

Reply:
That's right, better to say "I don't know" than lie about types :P Changed it here: https://github.com/confio/ts-relayer/pull/159/commits/bf99247f7cf609125b03c73919a9760f41cf8cb0